inference-engine/src/inference_engine/src/cpp/ie_infer_request.cpp

```diff
@@ -213,20 +214,51 @@ InferRequest::InferRequest(const std::shared_ptr<void>& so, const ie::IInferRequ
     OPENVINO_ASSERT(_impl != nullptr, "InferRequest was not initialized.");
 }
 
-void InferRequest::set_blob(const std::string& name, const ie::Blob::Ptr& data) {
-    OV_INFER_REQ_CALL_STATEMENT(_impl->SetBlob(name, data);)
+static std::string get_blob_name(const ie::IExecutableNetworkInternal::Ptr& exec_network, const std::string& name) {
+    IE_ASSERT(exec_network != nullptr);
+    auto func = exec_network->GetExecGraphInfo();
+    IE_ASSERT(func != nullptr);
+    for (const auto& parameter : func->get_parameters()) {
+        IE_ASSERT(parameter->get_output_size() == 1);
+        for (const auto& tensor_name : parameter->get_output_tensor(0).get_names()) {
+            if (tensor_name == name) {
+                return parameter->get_friendly_name();
+            }
+        }
+    }
+    for (const auto& result : func->get_results()) {
+        for (const auto& input : result->inputs()) {
+            auto output = input.get_source_output();
+            for (const auto& tensor_name : output.get_tensor().get_names()) {
+                if (tensor_name == name) {
+                    return ngraph::op::util::create_ie_output_name(output);
+                }
+            }
+        }
+    }
+    IE_THROW(NotFound) << "Tensor with name " << name << " was not found";
 }
 
-ie::Blob::Ptr InferRequest::get_blob(const std::string& name) {
-    ie::Blob::Ptr blobPtr;
-    OV_INFER_REQ_CALL_STATEMENT(blobPtr = _impl->GetBlob(name);)
-    std::string error = "Internal error: blob with name `" + name + "` is not allocated!";
-    const bool remoteBlobPassed = blobPtr->is<ie::RemoteBlob>();
-    if (blobPtr == nullptr)
-        IE_THROW() << error;
-    if (!remoteBlobPassed && blobPtr->buffer() == nullptr)
-        IE_THROW() << error;
-    return blobPtr;
+void InferRequest::set_tensor(const std::string& name, const Tensor& tensor){OV_INFER_REQ_CALL_STATEMENT({
+    std::string blob_name = get_blob_name(_impl->GetExecutableNetwork(), name);
+    _impl->SetBlob(blob_name, tensor._impl);
+})}
```

Comment:
can we remove these magic with names and use `name` as is? (or keep on internal branch)

Tensor names, etc will be supported by @ilyachur separately

Reply:
removed